src/ApnPrebidVast.js

```diff
@@ -0,0 +1,350 @@
+/**
+ * Header Bidding Plugin Brightcove module.
+ */
+
+// CONSTANTS
+
+var _vjs = window.videojs !== undefined ? window.videojs : null;
+var _vastManager = require('./VastManager.js');
+var _prebidCommunicator = require('./PrebidCommunicator.js');
+var _logger = require('./Logging.js');
+var _prefix = 'apnPrebidVast->';
+
+_logger.always(_prefix, 'Version 1.0.9');
+
+var APN_prebid_in_progress = window.apn_plugin_prebid_options && window.apn_plugin_prebid_options.biddersSpec;
```

Comment:
Should purge global, mutable state from the project. It's tough to test, and causes an enormous number of bugs.

There are quite a few of these floating around.

Reply:
We cannot avoid using global vars with the prebid plugin for certain cases. The global vars are needed when the publisher wants to run prebid in the header. The publisher in this case needs to define the bidder settings BEFORE the plugin is loaded.  When the plugin loads, then it first looks to see if these bidder settings have already been defined.  If they are, then 
the plugin can execute prebid in the header.  If they are not defined, then the plugin waits until the publisher invokes renderAd AFTER the Brightcove Player has been loaded.  These global variables allow the plugin to support both prebid in header and prebid in the body (just-in-time).  We have noted that prebid.js uses its own global var 'pbjs'. Furthermore, videojs itself, which is used for rendering the ad, also is a global. I think we can use same methodology that prebid.js uses: accomulate all global vars in one scope to avoid many global vars.
Is this sufficient for you?  

It would be inconvenient if we switch the version from 1.0.x to 0.x because the name of our branch is VIDLA-VER-1.0.9.  It would be confusing if the branch cited version 1.0.9 but the version cited in the code was 0.x.  We would prefer to keep version 1.0.9  in the current branch.   We will not merge this branch to master until we are sure the code has passed the code review and is ready for release.

Reply:
> We have noted that prebid.js uses its own global var 'pbjs'. Furthermore, videojs itself, which is used for rendering the ad, also is a global. I think we can use same methodology that prebid.js uses: accomulate all global vars in one scope to avoid many global vars. Is this sufficient for you?

Yes. Sorry... I misspoke. It is customary for each javascript script to have a single global... but more than that is considered bad practice because it pollutes the page's namepsace.

> It would be inconvenient if we switch the version from 1.0.x to 0.x because the name of our branch is VIDLA-VER-1.0.9.

You can rename a branch with `git branch -m new-name`, push the new branch, and open a new PR under that name...

`1.x` in open source code carries some very strong guarantees. You're no longer allowed to introduce breaking changes into the API unless you bump the major version, and publishers will be scared to update new major versions... so you're basically committing to support this API forever.

If you're ok with that, though, then I won't press the issue... I won't really be maintaining this code.

Reply:
Yes, I can rename the current branch to 0.x and open a new PR. But I do not know how to transfer comments from the current PR to new one. Is there a way to do that?